test/diameter_test_server.erl

```diff
@@ -276,30 +276,38 @@ handle_request(#diameter_packet{msg = ['AAR' | Msg]}, _SvcName, {_, Caps}, _Extr
     #{'Session-Id' := Id,
       'Auth-Request-Type' := Type,
       'Auth-Application-Id' := AppId} = Msg,
+    ct:pal("AAR received: ~p", [Msg]),
     FramedIP = case maps:get('Framed-IP-Address', Msg, [<<0,0,0,0>>]) of
 		   [<<0,0,0,0>>] -> [<<10,106,14,227>>];
 		   FramedIPReq -> FramedIPReq
 	       end,
-    AAA0 =  #{'Session-Id' => Id,
-	      'Result-Code' => 2001,
-	      'Origin-Host' => OH,
-	      'Origin-Realm' => OR,
-  	      'SN-Primary-DNS-Server' => [{1,2,3,4}],
-	      'SN-Secondary-DNS-Server' => [{5,6,7,8}],
- 	      'Acct-Interim-Interval' => [InterimAccounting],
-	      'Authorization-Lifetime' => [AuthLifeTime],
-	      'Framed-IP-Address' => FramedIP,
-	      'Framed-MTU' => 1500,
-	      'Auth-Request-Type' => Type,
-	      'Auth-Application-Id' => AppId},
-    AAA = if is_map_key('Framed-IPv6-Prefix', Msg) ->
-		  AAA0#{'3GPP-IPv6-DNS-Servers' =>
-			    ergw_aaa_3gpp_dict:encode('3GPP-IPv6-DNS-Servers',
-						      [{16#fe80,0,0,0,0,0,0,1},
-						       {16#fe80,0,0,0,0,0,0,2}])};
-	     true -> AAA0
-	  end,
-    {reply, ['AAA' | AAA]};
+    OkResponse = fun() ->
```

Comment:
why is that a fun? I don't see the need for that.

This could simply be:
```
OkResponse = {reply, ['AAA' | AAA]},
```

resulting in a much smaller change.

Reply:
I wanted to avoid doing all this processing unless it's really necessary, but maybe it's a premature optimization. If you prefer this way, I can change it.

Reply:
that code is used in unit test only, I really don't see that that would need optimization.

You could also move the answer construction into the final case when you change it a bit to

```Erlang
    case Msg of
	#{'Calling-Station-Id' := ?MSISDN_FOR_IMEI_SV}
	  when not is_map_key('3GPP-IMEISV') ->
	    {answer_message, 5005};
	_ ->
	    < code for ok response >
    end;
```

Reply:
Ok, I'll remove the fun and use the precomputed ok response.

Reply:
Re: the answer construction, I prefer using explicit conditions for separate cases in tests. The "when not is_map_key" requires that the reader spends time thinking on whether the case they're using complies or not with this condition. This may be ok for the code itself, but makes the life of a tester easier.